<a href="https://colab.research.google.com/github/Aldistwn29/intro-pandas/blob/main/intro_to_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Intro to Pandas

In [ ]:
import pandas as pd

mydataset = {
    "cars" : ["BMW", "Voivo", "Ford"],
    "Pasings": [3, 7, 2]
}

myvar = pd.DataFrame(mydataset)

print(myvar)

    cars  Pasings
0    BMW        3
1  Voivo        7
2   Ford        2


In [ ]:
import pandas as pd

print(pd.__version__)

1.5.3


# Pandas Series

In [ ]:
import pandas as pd

a = [1, 2, 3]

myvar = pd.Series(a)
print(myvar)

0    1
1    2
2    3
dtype: int64


In [ ]:
print(myvar[1])

2


In [ ]:
import pandas as pd

a = [1, 2, 3]

myvar = pd.Series(a, index =["X", "Y", "Z"])
print(myvar)

X    1
Y    2
Z    3
dtype: int64


Setelah membuat label, Kita Dapat mengakses item dengan merujuk label

In [ ]:
print(myvar["Z"])

3


# Key/Object as Series

In [ ]:
import pandas as pd

calories = {"day1": 420, "day2": 500, "day3": 300}

myvar = pd.Series(calories)
print(myvar)

day1    420
day2    500
day3    300
dtype: int64


Untuk memilih hanya beberapa item dalam kamus, gunakan indeks argumen dan hanya item itu yg akan di munculkan dalam seri

In [ ]:
import pandas as pd

calories = {"day1": 420, "day2": 500, "day3": 300}

myvar = pd.Series(calories, index = ["day2", "day3"])
print(myvar)

day2    500
day3    300
dtype: int64


# DataFrame

In [ ]:
import pandas as pd

data = {
    "calories": [420, 500, 300],
    "duration": [50, 40, 55]
}

myvar= pd.DataFrame(data)
df = pd.DataFrame(data)
print(myvar)

   calories  duration
0       420        50
1       500        40
2       300        55


# Locate Row

In [ ]:
# refer to the row index
print(df.loc[0])

calories    420
duration     50
Name: 0, dtype: int64


In [ ]:
# use a list of index:
print(df.loc[[0, 2]])

   calories  duration
0       420        50
2       300        55


# Name index

In [ ]:
import pandas as pd

data = {
    "calories": [420, 500, 300],
    "duration": [50, 40, 40]
}

df = pd.DataFrame(data, index = ["day1", "day2", "day3"])
print(df)

      calories  duration
day1       420        50
day2       500        40
day3       300        40


# Locate Named indexs

In [ ]:
# refer to the named index:
print(df.loc["day2"])

calories    500
duration     40
Name: day2, dtype: int64


# Load files to dataFame


In [ ]:
import pandas as pd

df = pd.read_csv('data.csv')
print(df.to_string())

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

In [ ]:
print(df)

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]


max_rows

Jumlah baris yg dikembalikan ditentukan dalam pengaturan opsi pandas. Kita dapat memeriksa baris maksimum sistem kita dengan pernyataan pd.options.display.max_rows.

In [ ]:
pd.options.display.max_rows = 9999

In [ ]:
print(df)

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

# Analyze DataFrame

In [ ]:
print(df.head(10))

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
5        60    102       127     300.0
6        60    110       136     374.0
7        45    104       134     253.3
8        30    109       133     195.1
9        60     98       124     269.0


In [ ]:
print(df.tail())

     Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


In [ ]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB
None


metode info() juga memberi tahu kita beberapa banyak nilai Non_Null yg ada di setiap kolom, dan dalam kumpulan data kita seperti ada 264 dari 169 nilai Non-Null di kolom "calories".

artinya ada 5 baris tanpa nilai sama sekali, di kolom "calories", untuk alasan apa pun.
Nilai Kosong, atau nilai Null, bisa berakibat buruk saat menganalisis data, dan kita harus mempertimbangkan untuk menghapus baris denhan nilai kosong. ini adalah langkah menuju apa yg disebut membersihkan data, dan kita akan mempelajarinnya lebih lanjut di bab-bab berikutnya